tests/system/test_system.py

```diff
@@ -2439,6 +2441,52 @@ def test_ubla_set_unset_preserves_acls(self):
         self.assertEqual(bucket_acl_before, bucket_acl_after)
         self.assertEqual(blob_acl_before, blob_acl_after)
 
+    def test_new_bucket_w_pap(self):
+        new_bucket_name = "new-w-pap" + unique_resource_id("-")
+        self.assertRaises(
+            exceptions.NotFound, Config.CLIENT.get_bucket, new_bucket_name
+        )
+        bucket = Config.CLIENT.bucket(new_bucket_name)
+        bucket.iam_configuration.uniform_bucket_level_access_enabled = True
+        bucket.create()
+        self.case_buckets_to_delete.append(new_bucket_name)
+
+        self.assertEqual(
+            bucket.iam_configuration.public_access_prevention,
+            PUBLIC_ACCESS_PREVENTION_UNSPECIFIED,
+        )
+
+        bucket.iam_configuration.public_access_prevention = (
+            PUBLIC_ACCESS_PREVENTION_ENFORCED
+        )
+        bucket.patch()
+        self.assertEqual(
+            bucket.iam_configuration.public_access_prevention,
+            PUBLIC_ACCESS_PREVENTION_ENFORCED,
+        )
+        self.assertTrue(bucket.iam_configuration.uniform_bucket_level_access_enabled)
+
+        bucket.iam_configuration.uniform_bucket_level_access_enabled = False
+        bucket.patch()
+        self.assertEqual(
+            bucket.iam_configuration.public_access_prevention,
+            PUBLIC_ACCESS_PREVENTION_ENFORCED,
+        )
+
+        with self.assertRaises(exceptions.BadRequest):
+            bucket.iam_configuration.public_access_prevention = "unexpected value"
+            bucket.patch()
+
+        with self.assertRaises(exceptions.PreconditionFailed):
+            bucket.make_public()
+
+        blob_name = "my-blob.txt"
+        blob = bucket.blob(blob_name)
+        payload = b"DEADBEEF"
+        blob.upload_from_string(payload)
+        with self.assertRaises(exceptions.PreconditionFailed):
+            blob.make_public()
+
```

Comment:
ISTM we need another systest for the case that the `public_access_prevention` flag is set to `"enforced"` before the bucket is created.  Also, do we need to test whether setting it back from `"enforced"` to `"unspecified"` works as intended?

Reply:
Good callout. I will implement these two tests cases.